packages/rum-react/src/get-with-transaction.js

```diff
@@ -40,6 +41,45 @@ function isReactClassComponent(Component) {
   return !!(prototype && prototype.isReactComponent)
 }
 
+const getMemoizedApmComponent = memoize((Component, apm, name, type) => {
```

Comment:
@dgieselaar WDYT about memoize a function that returns a Component? I'm not sure about it. I need to do it to avoid creating a new ApmComponent every time a change happens in the URL.

Reply:
Well, it's a memory leak I think. I would at least tie it to the lifecycle of a container (e.g. by using `getDerivedStateFromProps`). I think you also might be able to rewrite this to a component that takes `apm`, `name` and `type` as props, and then you don't need memoization at all.